From fcfa8310246b8ad8eaeade8d0cd2e49449a3a33c Mon Sep 17 00:00:00 2001
From: Ivan Kardykov <kardykov@tabit.pro>
Date: Thu, 9 Dec 2021 11:57:58 +0300
Subject: [PATCH] fix win32 compilation

---
 daemon/db-cmds.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/daemon/db-cmds.c b/daemon/db-cmds.c
index ba2f15b..67fae05 100644
--- a/daemon/db-cmds.c
+++ b/daemon/db-cmds.c
@@ -154,7 +154,9 @@ int write_vchan_or_client(struct db_daemon_data *d, struct client *c,
             return 1;
         write_queue = d->vchan_buffer;
     } else {
+#ifndef WIN32
         write_queue = c->write_queue;
+#endif
     }
 
 #ifdef WIN32
@@ -168,9 +170,11 @@ int write_vchan_or_client(struct db_daemon_data *d, struct client *c,
 
     /* now it's either vchan, or local client on Linux */
     while ((buf_datacount = buffer_datacount(write_queue))) {
+#ifndef WIN32
         if (c)
             ret = write(c->fd, buffer_data(write_queue), buf_datacount);
         else
+#endif
             ret = vchan_write_nonblock(d->vchan, buffer_data(write_queue), buf_datacount);
         if (ret < 0) {
             if (errno == EAGAIN || errno == EWOULDBLOCK) {
@@ -187,9 +191,11 @@ int write_vchan_or_client(struct db_daemon_data *d, struct client *c,
 
     count = 0;
     while (count < data_len) {
+#ifndef WIN32
         if (c)
             ret = write(c->fd, data+count, data_len-count);
         else
+#endif
             ret = vchan_write_nonblock(d->vchan, data+count, data_len-count);
         if (ret < 0) {
             if (errno == EAGAIN || errno == EWOULDBLOCK) {
-- 
2.31.1

