From 6501c5f826a78053c0cee8d03e0c6e516c0e53d5 Mon Sep 17 00:00:00 2001
From: Mikhail Lukashov <mlukashov@tabit.pro>
Date: Thu, 2 Dec 2021 16:50:32 +0300
Subject: [PATCH 1/2] qubes-windows-utils: LogReadLevel: minor improvements:
 remove compilation warning

---
 src/log.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/log.c b/src/log.c
index 67d5f53..c3c1e06 100644
--- a/src/log.c
+++ b/src/log.c
@@ -137,10 +137,13 @@ static WCHAR *LogGetName(void)
 static void LogReadLevel(void)
 {
     DWORD status;
-    status = CfgReadDword(LogGetName(), LOG_CONFIG_LEVEL_VALUE, &g_LogLevel, NULL);
+    DWORD logLevel;
+    status = CfgReadDword(LogGetName(), LOG_CONFIG_LEVEL_VALUE, &logLevel, NULL);
 
     if (status != ERROR_SUCCESS)
         g_LogLevel = LOG_LEVEL_INFO; // default
+    else
+        g_LogLevel = logLevel;
 }
 
 // Explicitly set verbosity level.
-- 
2.34.1.windows.1

