From 8d890b8e7896eea60a84fc79a15e5c83df7468e2 Mon Sep 17 00:00:00 2001
From: Mikhail Lukashov <mlukashov@tabit.pro>
Date: Thu, 2 Dec 2021 16:43:03 +0300
Subject: [PATCH] db-daemon: decrease log level; add more detailed log
 description

---
 daemon/db-daemon.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/daemon/db-daemon.c b/daemon/db-daemon.c
index b6c82f3..5474160 100644
--- a/daemon/db-daemon.c
+++ b/daemon/db-daemon.c
@@ -258,8 +258,7 @@ DWORD WINAPI pipe_thread_client(PVOID param) {
         // blocking read
         status = QpsRead(p->daemon->pipe_server, p->id, &hdr, sizeof(hdr));
         if (ERROR_SUCCESS != status) {
-            win_perror("QpsRead");
-            LogWarning("read from client %lu failed", p->id);
+            LogWarning("QpsRead from client %lu failed: %d", p->id, (int)status);
             QpsDisconnectClient(p->daemon->pipe_server, p->id);
             free(param);
             return status;
-- 
2.34.1.windows.1

