From afdb006f89399f6943c3c25c9f7b3d37fb07ec32 Mon Sep 17 00:00:00 2001
From: Mikhail Lukashov <mlukashov@tabit.pro>
Date: Mon, 6 Dec 2021 12:30:55 +0300
Subject: [PATCH] qubesdb-daemon: add xenagent dependency

---
 windows/core-qubesdb.wxs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/windows/core-qubesdb.wxs b/windows/core-qubesdb.wxs
index 93db7bd..5981133 100644
--- a/windows/core-qubesdb.wxs
+++ b/windows/core-qubesdb.wxs
@@ -62,7 +62,9 @@ Value='[=ProductFolder]'
                         ErrorControl='ignore'
                         Interactive='no'
                         Arguments='0'
-                        />
+                        >
+                        <ServiceDependency Id='xenagent' />
+                        </ServiceInstall>
 
                         <ServiceControl
                         Id='QdbServiceControl'
-- 
2.34.1.windows.1

