From f0d1abe85fc51cfa8906b156da37395713cfb612 Mon Sep 17 00:00:00 2001
From: Mikhail Lukashov <mlukashov@tabit.pro>
Date: Tue, 23 Nov 2021 11:45:48 +0300
Subject: [PATCH] qrexec-agent: call libvchan_close when service is stopped

---
 src/qrexec-agent/qrexec-agent.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/src/qrexec-agent/qrexec-agent.c b/src/qrexec-agent/qrexec-agent.c
index d04c59d..aa2464f 100644
--- a/src/qrexec-agent/qrexec-agent.c
+++ b/src/qrexec-agent/qrexec-agent.c
@@ -973,6 +973,8 @@ static DWORD HandleDaemonMessage(void)
     }
 }
 
+static DWORD WINAPI ServiceCleanup(void);
+
 /**
  * @brief Vchan event loop.
  * @param stopEvent When this event is signaled, the function should exit.
@@ -1087,8 +1089,7 @@ static DWORD WatchForEvents(HANDLE stopEvent)
 
     LogVerbose("loop finished");
 
-    if (daemonConnected)
-        libvchan_close(g_DaemonVchan);
+    ServiceCleanup();
 
     return status;
 }
@@ -1278,6 +1279,7 @@ DWORD WINAPI ServiceExecutionThread(void *param)
     if (ERROR_SUCCESS != status)
         win_perror2(status, "WatchForEvents");
 
+    ServiceCleanup();
 // FIXME: pipe server doesn't have the ability for graceful stop
 
     LogDebug("Waiting for the pipe server thread to exit");
@@ -1293,6 +1295,16 @@ DWORD WINAPI ServiceExecutionThread(void *param)
     return ERROR_SUCCESS;
 }
 
+static DWORD WINAPI ServiceCleanup(void)
+{
+    if (g_DaemonVchan)
+    {
+        libvchan_close(g_DaemonVchan);
+        g_DaemonVchan = NULL;
+    }
+    return ERROR_SUCCESS;
+}
+
 int __cdecl wmain(int argc, WCHAR *argv[])
 {
     DWORD status;
-- 
2.34.0.windows.1

